### PR TITLE
twister: filters: add ability to filter by vendor

### PR DESCRIPTION
**Pull request by nashif**
add vendor_allow and vendor_exclude to the list of filters. The vendor
data is maintained in the board.yml files.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: scripts/pylib/twister/twisterlib/testplan.py [hunk 1]
```diff
@@ -895,6 +895,12 @@ def apply_filters(self, **kwargs):
                     if ts.arch_exclude and plat.arch in ts.arch_exclude:
                         instance.add_filter("In test case arch exclude", Filters.TESTSUITE)
 
+                    if ts.vendor_allow and plat.vendor not in ts.vendor_allow:
```
Comment (golowanow) [inline scripts/pylib/twister/twisterlib/testplan.py:898]: when `plat.vendor` is empty (not specified) the platform will be excluded - It looks correct, but better to document it for the new test attribute.

File: scripts/pylib/twister/twisterlib/testplan.py [hunk 2]
```diff
@@ -895,6 +895,12 @@ def apply_filters(self, **kwargs):
...
+                        instance.add_filter("Not in test suite vendor allow list", Filters.TESTSUITE)
+
+                    if ts.vendor_exclude and plat.vendor in ts.vendor_exclude:
```
Comment (golowanow) [inline scripts/pylib/twister/twisterlib/testplan.py:901]: also has a trick with `plat.vendor`: when in testcase.yaml for some mistake/reason there is an empty `vendor_exclude` item, then all platforms without vendor specified will be excluded.

Comment (nashif) [reply scripts/pylib/twister/twisterlib/testplan.py:901]: if there is an empty `vendor_exclude`, we will not get into this conditional.

File: scripts/pylib/twister/twisterlib/testplan.py (hunk 1 above)
Comment (nashif) [reply scripts/pylib/twister/twisterlib/testplan.py:898]: pedantic, but sure, why not.

File: scripts/pylib/twister/twisterlib/testplan.py (hunk 2 above)
Comment (golowanow) [reply scripts/pylib/twister/twisterlib/testplan.py:901]: hmm, I've tried it this way and it was filtered:
```
vendor_exclude:
 -
```

Comment (nashif) [reply scripts/pylib/twister/twisterlib/testplan.py:901]: well, you are adding an empty element, so there is a list with an empty element. The same "mistake" can happen anywhere, it is not something unique to vendor_exclude, try this with other keys and you will get different behaviours all over the place.

**Review by golowanow — APPROVED** [review]

**Review by hakehuang — APPROVED** [review]

**Pull request merged**